dtests/src/test/java/io/snappydata/hydra/cluster/SnappyTest.java

```diff
@@ -0,0 +1,756 @@
+        } catch (Exception e) {
+            throw new TestException("Following error occurred while executing " + e.getMessage());
+        }
+        return scriptPath;
+    }
+
+    protected String getUserDataLocation(String scriptName) {
+        String scriptPath = null;
+        try {
+            scriptPath = hd.getGemFireHome() + sep + ".." + sep + ".." + sep + ".." + sep + "dtests" + sep + "src" + sep + "resources" + sep + "data" + sep + scriptName;
+        } catch (Exception e) {
+            throw new TestException("Following error occurred while executing " + e.getMessage());
+        }
+        return scriptPath;
+    }
+
+    protected String getUserAppJarLocation(String userAppJar) {
+        String jarPath = null;
+        try {
+            jarPath = hd.getGemFireHome() + sep + ".." + sep + ".." + sep + ".." + sep + "dtests" + sep + "build-artifacts" + sep + "scala-2.10" + sep + "libs" + sep + userAppJar;
+        } catch (Exception e) {
+            throw new TestException("Following error occurred while executing " + e.getMessage());
+        }
+        return jarPath;
+    }
+
+    public static void HydraTask_initializeSnappyTest() {
+        if (snappyTest == null) {
+            snappyTest = new SnappyTest();
+            snappyTest.generateConfig("locators");
+            snappyTest.generateConfig("servers");
+            snappyTest.generateConfig("leads");
+        }
+    }
+
+    public static synchronized void HydraTask_generateSnappyConfig() {
+        locatorsFilePath = snappyTest.getSnappyPath() + sep + "conf" + sep + "locators";
+        serversFilePath = snappyTest.getSnappyPath() + sep + "conf" + sep + "servers";
+        leadsFilePath = snappyTest.getSnappyPath() + sep + "conf" + sep + "leads";
+        try {
+            String dirPath = snappyTest.getLogDir();
+            File locatorsFile = new File(locatorsFilePath);
+            File serversFile = new File(serversFilePath);
+            File leadsFile = new File(leadsFilePath);
+            if (dirPath.contains("locator")) {
+                String locatorLogDir = localHost + " -dir=" + dirPath;
+                if (!locatorsFileContent.contains(locatorLogDir)) {
+                    locatorsFileContent.add(locatorLogDir);
+                    snappyTest.writeToFile(locatorLogDir, locatorsFile);
+                }
+            } else if (dirPath.contains("Store") || dirPath.contains("server")) {
+                String serverLogDir = localHost + " -dir=" + dirPath;
+                if (!serversFileContent.contains(serverLogDir)) {
+                    serversFileContent.add(serverLogDir);
+                    snappyTest.writeToFile(serverLogDir, serversFile);
+                }
+            } else if (dirPath.contains("lead")) {
+                String leadLogDir = localHost + " -dir=" + dirPath;
+                if (leadHost == null) {
```

Comment:
This variable is required at other places in my code. I am setting its value here and using that value at other locations. e.g. executeSnappyJob method.